### ISSUE TITLE
Extend is_numeric_linter() to look for approaches checking class() or typeof() or inherits() equivalent to is_numeric()

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2470]
First, try and gauge how common this is in practice. Then try and extend the XPath to cover these. Some examples:

```r
any(class(x) %in% c("integer", "numeric")) # bad use of class() should be linted by class_equals_linter()
typeof(x) %in% c("integer", "double")
inherits(x, c("integer", "numeric"))
```
